vtm/resources/assets/shaders/linetex_layer_tex.glsl

```diff
@@ -44,15 +44,19 @@ uniform float u_mode;
 void
 main(){
   if (u_mode >= 1.0) {
+    /* Dash array or texture */
     float step = 2.0;
     if (u_mode == 2.0) { // dashed texture
       step = 1.0;
     }
-    vec4 c = texture2D(tex, vec2(abs(mod(v_st.s + 1.0, step)), (v_st.t + 1.0) * 0.5));
+    // use linelength mod texture step (mod is always positiv)
+    vec4 c = texture2D(tex, vec2(mod(v_st.s, step), (v_st.t + 1.0) * 0.5));
```

Comment:
Why the remove of the "+ 1.0" in mod?

Reply:
The better question is, why there was a "+ 1.0" ?
`v_st.s` is the interpolated lineLength value between two vertices, which is divided with `u_pscale`.
Theoretically `v_st.s` can have a range from `-∞` to `+∞`. So it makes no sense to add 1 as it is taken modulo anyway and then has (dependent on dash flag) a range from `0` to `1` or `2`.
The only reason could be that texture is shifted so it looks better at the beginning, but I didn't noticed any better look. If leaving `+ 1.0` we should comment that it has no further meaning.

Reply:
Seems the answer is found in #536. :slightly_smiling_face: